client/blocks/login/social.jsx

```diff
@@ -25,6 +25,27 @@ import {
 import { recordTracksEvent } from 'state/analytics/actions';
 import WpcomLoginForm from 'signup/wpcom-login-form';
 import { InfoNotice } from 'blocks/global-notice';
+import { login } from 'lib/paths';
+
+function isLocalStorageFunctional() {
+	try {
+		window.localStorage.setItem( '__test_localStorage', 1 );
+		window.localStorage.removeItem( '__test_localStorage' );
+		return true;
+	} catch ( error ) {
+		return false;
+	}
+}
+
+function shouldUseRedirectFlow() {
+	// If calypso is loaded in a popup, we don't want to open a second popup for social login
+	// let's use the redirect flow instead in that case
+	const isPopup = typeof window !== 'undefined' && window.opener && window.opener !== window;
+	// also disable the popup flow for old versions of safari private browsing (<v11)
+	// See https://github.com/google/google-api-javascript-client/issues/297#issuecomment-333869742
```

Comment:
I don't think that's the proper check, 
https://jsfiddle.net/d9h2Lya4/

Returns `true` in safari private mode, yet the popup is failing.
TMCH mentioned it's not related to localStorage at the thread over at `google-api-javascript-client`

Reply:
Yeah because it was fixed in the last version of Safari I think. We can hardly detect private browsing in Safari now but I'm not sure we should use the redirect flow for all safari version, should we?